pybop/problems/fitting_problem.py

```diff
@@ -18,12 +19,25 @@ class FittingProblem(BaseProblem):
         An object or list of the parameters for the problem.
     dataset : Dataset
         Dataset object containing the data to fit the model to.
+    check_model : bool, optional
+        Flag to indicate if the model should be checked (default: True).
     signal : str, optional
         The variable used for fitting (default: "Voltage [V]").
     additional_variables : List[str], optional
         Additional variables to observe and store in the solution (default additions are: ["Time [s]"]).
     init_soc : float, optional
         Initial state of charge (default: None).
+
+    Additional Attributes
+    ---------------------
+    _dataset : dictionary
+        The dictionary from a Dataset object containing the signal keys and values to fit the model to.
+    _time_data : np.ndarray
+        The time points in the dataset.
+    n_time_data : int
+        The number of time points.
+    _target : np.ndarray
+        The target values of the signals.
```

Comment:
Here's a picky comment, but in PyBaMM we had a similar discussion a while ago: do we want to document methods and attributes meant for internal use? I believe it PyBaMM we decided not to, as users are not meant to use them, but there is good reasons to go either way.

Reply:
That's an interesting decision! Let's include concise documentation for all attributes and methods in PyBOP. This information is vital for new developers.